### PR TITLE
Workflow update

### PR DESCRIPTION
**Pull request by bogatykh**
Do we need 18.04 Ubuntu to build OpenJDK?

### CODE REVIEW

**Comment by wasabii** [comment]
We did when I originally put this together. The Lenny distro uses syscalls not available in later kernel versions, if I remember.

**Comment by wasabii** [comment]
You'll have to make sure the cache is expired, or it won't even try building, though. The IKVM.yml line for the name of the NuGet cache has a number at the end that you can incremement.

**Comment by bogatykh** [comment]
> You'll have to make sure the cache is expired, or it won't even try building, though. The IKVM.yml line for the name of the cache has a number at the end that you can incremement.

OK I will check it out. Anyway we will have to figure out what to do with it due to depreciation 
At least we need to update github actions to latest versions, this looks safe to me.

**Comment by wasabii** [comment]
Yeah the versions are fine.

Gotta get the JDK upgraded. But it's such a slow process. The latest jdk8 builds fine.

**Comment by bogatykh** [comment]
@wasabii looks like intermittent failure of unit test. Can we rerun workflow?

**Comment by wasabii** [comment]
@bogatykh You'll have to make that another commit to expire the OpenJDK build cache. Else we don't know whether it actually builds.

I have it cached because it takes a significant amount of time, and it's not like we ever change it.

**Comment by bogatykh** [comment]
@wasabii you may delete cached entry from web with write permissions (I don't have them).
Take a look: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#deleting-cache-entries

I can make commit later to expire cache if above instructions won't work for you.

**Comment by wasabii** [comment]
You'll need to do it in your branch to generate a new cache entry for your branch (encapsulating your changes).

**Comment by bogatykh** [comment]
> You'll need to do it in your branch to generate a new cache entry for your branch (encapsulating your changes).

Never done that.
Can you please verify my last commit?

**Comment by wasabii** [comment]
All you got to do is increment the integer at the end.

File: .github/workflows/IKVM.yml [hunk 1]
```diff
@@ -15,20 +15,20 @@ on:
...
     - name: Checkout Source
-      uses: actions/checkout@v1
+      uses: actions/checkout@v3
       with:
         submodules: recursive
     - name: Cache OpenJDK Build
       uses: actions/cache@v3
       with:
         path: openjdk/build/linux-x86_64-normal-server-release
-        key: openjdk-build-linux-x86_64-normal-server-release--${{ runner.os }}--${{ hashFiles('openjdk/**', '!openjdk/build') }}-8
```
Comment (wasabii) [inline .github/workflows/IKVM.yml:28]: All you need to do is put -9 on the end of it, not change the prefix.

**Comment by wasabii** [comment]
All that said, the latest Ubuntu does appear to work. Good news.

File: .github/workflows/IKVM.yml (hunk 1 above)
Comment (bogatykh) [reply .github/workflows/IKVM.yml:28]: Done. Thank you!

**Pull request merged**